quill-test/src/main/scala/test/Test.scala

```diff
@@ -0,0 +1,24 @@
+package test
+
+import io.getquill.{ PostgresDialect, SnakeCase, SqlMirrorContext }
+
+object Test {
+  val ctx = new SqlMirrorContext(PostgresDialect, SnakeCase)
+
+  import ctx._
+
+  case class MyTable(id: Int, name: String)
+
+  def main(args: Array[String]): Unit = {
+
+    implicit class RichMyTableQuery(q: Query[MyTable]) {
+      def sortByCols(cols: String) = quote {
+        infix"$q ORDER BY $cols".as[Query[MyTable]]
+      }
+    }
+
+    def runSortByCols(cols: String*) = run(query[MyTable].sortByCols(spliceQuery(cols.mkString(", "))))
+    println(runSortByCols("name", "age").string)
+    println(runSortByCols("name").string)
```

Comment:
sbt output is:
```
[info] quill/quill-test/src/main/scala/test/Test.scala:20:43: SELECT x.id, x.name FROM (SELECT x.* FROM my_table x ORDER BY ${cols.mkString(", ")}) x
[info]     def runSortByCols(cols: String*) = run(query[MyTable].sortByCols(spliceQuery(cols.mkString(", "))))
[info]                                           ^
[info] Done compiling.
[info] Running test.Test 
SELECT x.id, x.name FROM (SELECT x.* FROM my_table x ORDER BY name, age) x
SELECT x.id, x.name FROM (SELECT x.* FROM my_table x ORDER BY name) x
```
Maybe we should open an issue to normalize infix here so it wont wrap it in nested query when `Map` function is the same